Connect products report page to the data source of report API

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

Closes #244 . This PR relies on the `feature/242-244-chart-section` branch of #568 

- **CompareProductsTableCard**
    - align to data schema of report API
    - handle loading and empty status
    - some props adjustments
    - apply store's currency setting to format values
- **SummarySection**
    - align to data schema of report API
    - handle loading status
    - Move to `.~/reports` folder as a shared component
- Connect products report page to the data source of report API
- Remove mocked products report data

### Screenshots:

![Kapture 2021-05-11 at 11 01 48](https://user-images.githubusercontent.com/17420811/117751994-9c069a80-b248-11eb-939f-449ab9d24333.gif)

### Detailed test instructions:

💡. For easier mock testing, I pushed [another branch with some mock code](https://github.com/woocommerce/google-listings-and-ads/compare/feature/244-connect-data...feature/244-products-report-mock-local), which located at:

- Mock paid data
    https://github.com/woocommerce/google-listings-and-ads/blob/aaea8311f05fd624969ed42a871e82eb4638cf64/src/API/Site/Controllers/Ads/ReportsController.php#L84-L230
- Mock free data
    https://github.com/woocommerce/google-listings-and-ads/blob/aaea8311f05fd624969ed42a871e82eb4638cf64/src/API/Site/Controllers/MerchantCenter/ReportsController.php#L83-L197

🩺 . Test with the UI

1. Checkout `244-products-report-mock-local` branch if want to test with local mocked data
2. Head to the Products Report page `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Freports%2Fproducts`
3. Use functions on this page, e.g. filers, sorting, it should fetch data from report API if the report query hasn't loaded before
4. All data displayed on this page should be sourced from report API

eason9487 committed:
CompareProductsTableCard - align to data schema of report API

eason9487 committed:
CompareProductsTableCard - handle loading and empty status, and some props adjustments

eason9487 committed:
CompareProductsTableCard - apply store's currency setting to format values

eason9487 committed:
SummarySection - align to data schema of report API and handle loading status

eason9487 committed:
Connect products report page to the data source of report API

eason9487 committed:
Remove mocked products report data

eason9487 committed:
Move SummarySection to upper folder as a shared component

eason9487 committed:
Update relevant importing paths of SummarySection

tomalec reviewed (approved):
Tested locally, reviewed the code. Works and LGTM.
Left few notes for potential further improvements, to be addressed here or later.

eason9487 committed:
Disconnect the linking of selected metric and sorting parameter

Addressed comments

- https://github.com/woocommerce/google-listings-and-ads/pull/569#issuecomment-837067394
- https://github.com/woocommerce/google-listings-and-ads/pull/569#issuecomment-837054531

eason9487 committed:
Only pass needed data to SummarySection, ChartSection, and CompareProductsTableCard

Addressed comments

- https://github.com/woocommerce/google-listings-and-ads/pull/569#discussion_r629646888
- https://github.com/woocommerce/google-listings-and-ads/pull/569#discussion_r629647591

eason9487 committed:
Shouldn't tick the "select all" checkbox if no any product

eason9487 committed:
Fix render bugs if errors occur when fetching report API

tomalec reviewed (approved):
Re-reviewed the code, and tested locally. Checked also additional fixes for select all. LGTM.

eason9487 committed:
Use precise JSDoc and function name for getFreeFieldFromQuery function

Addressed comment https://github.com/woocommerce/google-listings-and-ads/pull/569#discussion_r630201207

eason9487 committed:
Update source code references
